Store/DAL/Interfaces/IRepository.cs

```diff
@@ -0,0 +1,14 @@
+﻿using DAL.Entities;
+
+namespace DAL.Interfaces
+{
+    public interface IRepository
```

Comment:
Add more details to the name, Repository of what? (Product)
Rename as IProductRepository